test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/AbstractClientYamlTestFragmentParserTestCase.java

```diff
@@ -39,4 +40,63 @@ public void tearDown() throws Exception {
     protected NamedXContentRegistry xContentRegistry() {
         return ExecutableSection.XCONTENT_REGISTRY;
     }
+
+    protected static SkipSectionContext versionOnlyContext(Version version) {
+        return new SkipSectionContext() {
+            @Override
+            public boolean clusterIsRunningOs(String osName) {
+                return false;
+            }
+
+            @Override
+            public boolean clusterHasFeature(String featureId) {
+                return true;
+            }
+
+            @Override
+            public boolean clusterVersionInRange(VersionRange range) {
+                return range.contains(version);
+            }
+
+        };
+    }
+
+    protected static SkipSectionContext neverSkipContext() {
+        return new SkipSectionContext() {
+            @Override
+            public boolean clusterIsRunningOs(String osName) {
```

Comment:
Couldn't these all be default implementations?

Reply:
Yes, I thought about that. Not a big fan of default interface implementations, but sure, it can be done that way.

Reply:
After removing SkipSectionContext and using mocks as @mosche suggested, this is no longer needed